File: src-tauri/src/entities/engine_state.rs
use crate::entities::{Container, DockerInfo, EngineStatus, Image, Network, Volume};
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

/// Centralized, serializable view of the Docker engine state.
#[derive(Debug, Serialize, Deserialize, Clone, Default)]
pub struct EngineState {
    // Core collections
    pub containers: HashMap<String, Container>,
    pub images: HashMap<String, Image>,
    pub volumes: HashMap<String, Volume>,
    pub networks: HashMap<String, Network>,

    // Engine state information
    pub engine_status: EngineStatus,
    pub docker_info: Option<DockerInfo>,

    // Metadata
    pub version: u64,
    pub last_updated: DateTime<Utc>,
}

File: src/types/engine-state.ts
import { Container } from '../components/containers/container-types';
import { Image } from '../components/images/image-types';
import { Network } from '../components/networks/network-types';
import { Volume } from '../components/volumes/volume-types';
import { EngineStatus } from './engine-status';
import { DockerInfo } from './docker-info';

// PruneResult type matching backend
export interface PruneResult {
  deleted: string[];
  space_reclaimed: number;
}

// EngineState interface matching backend exactly
export interface EngineState {
  // Core collections
  containers: Record<string, Container>;
  images: Record<string, Image>;
  volumes: Record<string, Volume>;
  networks: Record<string, Network>;

  // Engine state information
  engine_status: EngineStatus;
  docker_info: DockerInfo | null;

  // Metadata
  version: number;
  last_updated: string; // ISO string from DateTime<Utc>
}
